experiments/tree_tensor_network/ttn_1d_uniform.py

```diff
@@ -11,67 +11,121 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""1-dimensional uniform binary tree tensor network."""
+"""1-dimensional uniform binary tree tensor network.
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-import sys
-import copy
-import math
-import time
-import numpy as np
-import scipy.linalg as spla
-
-import tensorflow as tf
-
-import tensornetwork
-"""
 Index ordering conventions:
-
+```
 iso_012:
 
   0
   |
 (iso)
  / \
 1   2
-
+```
 
 iso_021:
-
+```
   0
   |
 (iso)
  / \
 2   1
+```
 """
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import sys
+import copy
+import math
+import time
+import contextlib
+import tensornetwork
+
+
+def set_backend(backend_name):
+  """Sets the backend to use for tree tensor network computations.
+
+  A backend must be set after importing the module.
+
+  Args:
+    backend: Possible values are "tensorflow", "jax", and "numpy".
+  """
+  global backend
```

Comment:
You can avoid this last global by making it a module object. 

See https://github.com/google/TensorNetwork/blob/master/tensornetwork/config.py
and https://github.com/google/TensorNetwork/blob/b6ac4a08b52b6c78d1169231d6c0ec06517dd25d/tensornetwork/__init__.py#L10 for an example on how to do it.

Reply:
Once this is fixed we can merge this code.

Reply:
I don't really understand the advantage of this; an extra file to avoid the use of a single "global" statement? I think the "global" statement is easier to understand. Also, this would seem to require prepending all `backend.something()` calls with a module name e.g. `config.backend.something()`. Did I miss something?

Reply:
Ok, I figured out a way of avoiding that issue. Still not convinced of the evil of "global" here :)